Lib/subprocess.py

```diff
@@ -1879,3 +1879,24 @@ def kill(self):
             """Kill the process with SIGKILL
             """
             self.send_signal(signal.SIGKILL)
+
+        @contextlib.contextmanager
+        def kill_on_error(self):
+            """Using context manager to kill subprocess if
```

Comment:
How does using a context manager kill the process and why would someone reading this understand that to be what it does?  Does it matter to the docstring reader if a context manager is used?

similar to the documentation comment: if an python exception is raised is non-specific.  raised from where and when?  this could easily be misinterpreted to have something to do with the exit status of the child process or even exceptions raised within it.

"on error" is a very ambiguious method name.  someone reading code calling this won't intuitively understand what it does other than hopefully kill the process.

Reply:
Using context manager can use `try catch` statement to trap the error. I udpated the docstring too.
I use on_os_error to replace on_error.